---
title: "Mouse_QC"

output: 
  html_document:
    toc: true
    toc_float: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Introduction

This is an R Markdown document for the Quality Control (QC) of Mammalian array data for Dementia Mouse samples. 


###Study Information
**Study:** Dementia Mouse 

**Description:** 2 plates of mouse samples (whole and sorted)  

**Arrays ran by:** Isabel Castanho, The University of Exeter Medical School

**Array used:** Mammalian Array 

**QC done by:**Aisha Dahir, The University of Exeter Medical School

**Date of QC:** `r format(Sys.Date(), format="%d %B %Y")`

**Sample tissue:** Cortex Entorhinalis, Hippocampus, Sorted nuclei (neuN_pos, neuN_neg and total)


```{r, warning=FALSE, message=FALSE, echo = FALSE}
library(tidyverse)
library(minfi)
library(HorvathMammalMethylChip40manifest)
library(HorvathMammalMethylChip40anno.test.unknown)

library(wateRmelon)
library(ggplot2)
library(gdata)
require(gridExtra)
library(grid)
```

# 1  Loading data

As the array is not 450K nor EPIC, reading in the files may differ as well as the following QC methods. Probes and samples will be excluded at the end of the sample.

```{r read files in , warning = FALSE, echo = FALSE, message = FALSE}
load("/gpfs/ts0/scratch/and202/Mouse/RGsetAll.rdat") #contains no fully methylated samples
sample_sheet <- read.csv("/gpfs/ts0/scratch/and202/Mouse/SampleSheetAgeN43version1.csv", stringsAsFactors = F, header = T)
probe_mapping_file <- read_csv("HorvathMammalChip_SpeciesGenomeCoordsUnique_v1.0.csv", col_types = cols(.default = "c"))
```

The phenotype file containing the sample information was loaded into R. There are `r nrow(sample_sheet)` samples of which `r nrow(sample_sheet$ExternalSampleID == "methylated_control")` samples are fully methylated control samples.


```{r make RGset, echo = FALSE, warning = FALSE, error = FALSE}
Mset = preprocessRaw(RGset)
```


# 2 Intensity check

The intensity check is the biggest indicator of sample quality. The median methylated signal intensity and unmethylated signal intensity for each sample is calculcated.

```{r medianintensities, include=FALSE, echo = FALSE}
m_intensities<-getMeth(Mset)
u_intensities<-getUnmeth(Mset)
M.median<-apply(m_intensities, 2, median)
U.median<-apply(u_intensities, 2, median)
QCmetrics<-cbind(sample_sheet,M.median, U.median)

sample_sheet$Control <- sample_sheet$ExternalSampleID == "methylated_control"
```

A histogram and scatter plot of the resulting data are plotted to visualise the data quality. Samples are coloured by methylation plate or institute, to make sure there are no batch effects.

```{r plotintensities, echo=FALSE}
# coloured by plate
plotfactor<-factor(QCmetrics$PlateNumber, levels=c(unique(QCmetrics$PlateNumber))) 
par(mfrow = c(1,2))
hist(M.median, xlab = "Median M intensity", main="Histogram of Median Methylated Intensities", cex.main=0.7)
hist(U.median, xlab = "Median U intensity", main="Histogram of Median Unmethylated Intensities", cex.main=0.7)
par(mfrow = c(1,1))
plot(M.median, U.median, pch = 16, xlab = "Median M intensity", ylab = "Median U intensity", col = rainbow(nlevels(plotfactor))[factor(plotfactor)], main="Scatter plot of Signal Intensities coloured by Plate")
par(xpd=TRUE)
legend("topright", levels(factor(plotfactor)), col = rainbow(nlevels(plotfactor)), pch = 10, cex=0.5)

# coloured by Tissue
plotfactor<-factor(QCmetrics$Tissue, levels=c(unique(QCmetrics$Tissue))) 
par(mfrow = c(1,1))
plot(M.median, U.median, pch = 16, xlab = "Median M intensity", ylab = "Median U intensity", col = rainbow(nlevels(plotfactor))[factor(plotfactor)], main="Scatter plot of Signal Intensities coloured by Tissue")
par(xpd=TRUE)
legend("topright", levels(factor(plotfactor)), col = rainbow(nlevels(plotfactor)), pch = 10, cex=0.5)

# coloured by AD Model
plotfactor<-factor(QCmetrics$AD_model, levels=c(unique(QCmetrics$AD_model))) 
par(mfrow = c(1,1))
plot(M.median, U.median, pch = 16, xlab = "Median M intensity", ylab = "Median U intensity", col = rainbow(nlevels(plotfactor))[factor(plotfactor)], main="Scatter plot of Signal Intensities coloured by AD_Model")
par(xpd=TRUE)
legend("topright", levels(factor(plotfactor)), col = rainbow(nlevels(plotfactor)), pch = 10, cex=0.5)

# coloured by Genotype
plotfactor<-factor(QCmetrics$Genotype, levels=c(unique(QCmetrics$Genotype))) 
par(mfrow = c(1,1))
plot(M.median, U.median, pch = 16, xlab = "Median M intensity", ylab = "Median U intensity", col = rainbow(nlevels(plotfactor))[factor(plotfactor)], main="Scatter plot of Signal Intensities coloured by Genotype")
par(xpd=TRUE)
legend("topright", levels(factor(plotfactor)), col = rainbow(nlevels(plotfactor)), pch = 10, cex=0.5)
```



### Plate 1

```{r reform samplesheet, echo = FALSE, warning= FALSE, message = FALSE}

########## Reformat the QCmetric chips
QCmetrics <- separate(data = QCmetrics, col = Basename, into = c("Chip_ID", "Chip_Position"), sep = "_")

QCmetrics$Chip_Row <- substring(QCmetrics$Chip_Position,1,3)
QCmetrics$Chip_Column <- substring(QCmetrics$Chip_Position,4,6)

QCmetrics$Chip_Position<-factor(QCmetrics$Chip_Position)
QCmetrics$Chip_ID<-factor(QCmetrics$Chip_ID, levels=rev(unique(QCmetrics$Chip_ID))) #keeps the levels of the factor in current order rather than sorting numerically/alphabetically, also reverses this order as heatmaps plot bottom to top



plates<-unique(QCmetrics$PlateNumber)

control_sentrix <- as.character(QCmetrics[which(QCmetrics$ExternalSampleID == "methylated_control"), "Chip_ID"])

#extract the legend (using a function found online)
g_legend<-function(a.gplot){
    tmp <- ggplot_gtable(ggplot_build(a.gplot))
    leg <- which(sapply(tmp$grobs, function(x) x$name) == "guide-box")
    legend <- tmp$grobs[[leg]]
    legend
}
```

```{r plate1, fig.height= 10, fig.width= 18,echo = FALSE,fig.show='hold',out.height="50%"}

samples<-QCmetrics[which(QCmetrics$PlateNumber == plates[1]),]
Sentrix <- as.character(unique(samples$Chip_ID))

plate2Meth <- list()
plate2UnMeth <- list()
for (i in 1:length(as.character(unique(samples$Chip_ID)))) {
  Sentrix_i <- Sentrix[i]
  Sample_Sentrix <- samples[which(samples$Chip_ID == Sentrix_i),]

  if (Sentrix_i == unique(control_sentrix)){
    control <- Sample_Sentrix[which(Sample_Sentrix$ExternalSampleID == "methylated_control"),]
    control$Control <- control$ExternalSampleID == "methylated_control"
    
    plateHeatmap <- ggplot(data=Sample_Sentrix, aes(y=Chip_Row, x=Chip_Column)) +
    scale_fill_gradientn(colours=colorRamps::matlab.like(100), limits=c(min(QCmetrics$U.median),max(QCmetrics$M.median))) +
    labs(x="", y="") +
    theme_minimal() + 
    coord_equal(clip = 'off') +
    theme(plot.title = element_text(hjust = 0.5), axis.text.x = element_text(angle = 0, hjust=1),
          axis.text.y = element_text(angle = 1, hjust=1),
          plot.margin = unit(c(0,15,0,0), "lines"))
     
  plate2Meth[[i]] <- plateHeatmap +
    geom_tile(aes(fill=M.median), colour = "white") +
    theme(legend.position = "none")  +
    geom_point(data=control, aes(y=Chip_Row, x=Chip_Column)) +
    annotate(geom = "text", x = 0.8, y = 7,
             label = Sentrix[i], hjust = 0, angle = 0, size = 5) 
  
  plate2UnMeth[[i]] <- plateHeatmap +
    geom_tile(aes(fill=U.median), colour = "white") +
    geom_point(data=control, aes(y=Chip_Row, x=Chip_Column)) +
    theme(legend.position = "none")  +
    annotate(geom = "text", x = 0.8, y = 7,
             label = Sentrix[i], hjust = 0, angle = 0, size = 5) 
  
  legendplot<-plateHeatmap + 
    geom_tile(aes(fill=U.median), colour = "white") +
    labs(fill="Intensity", alpha="Control") +
    geom_point(data=control, aes(y=Chip_Row, x=Chip_Column, alpha=control$Control)) +
    scale_alpha_manual(values=c(1,1,1)) + 
    guides(alpha = guide_legend(override.aes = list(colour="black", pch=16)))
  
    legend <- g_legend(legendplot)
} 
  else
  
  {plateHeatmap <- ggplot(data=Sample_Sentrix, aes(y=Chip_Row, x=Chip_Column)) +
    scale_fill_gradientn(colours=colorRamps::matlab.like(100), limits=c(min(QCmetrics$U.median),max(QCmetrics$M.median))) +
    labs(x="", y="") +
    theme_minimal() + 
    coord_equal(clip = 'off') +
    theme(plot.title = element_text(hjust = 0.5), axis.text.x = element_text(angle = 0, hjust=1),
          axis.text.y = element_text(angle = 1, hjust=1),
          plot.margin = unit(c(0,15,0,0), "lines"))
     
  plate2Meth[[i]] <- plateHeatmap +
    geom_tile(aes(fill=M.median), colour = "white") +
    theme(legend.position = "none")  +
    annotate(geom = "text", x = 0.8, y = 7,
             label = Sentrix[i], hjust = 0, angle = 0, size = 5) 
  
  plate2UnMeth[[i]] <- plateHeatmap +
    geom_tile(aes(fill=U.median), colour = "white") +
    theme(legend.position = "none")  +
    annotate(geom = "text", x = 0.8, y = 7,
             label = Sentrix[i], hjust = 0, angle = 0, size = 5) 
    
    legendplot<-plateHeatmap + 
    geom_tile(aes(fill=U.median), colour = "white") +
    labs(fill="Intensity", alpha="Control") +
    scale_alpha_manual(values=c(1,1,1)) + 
    guides(alpha = guide_legend(override.aes = list(colour="black", pch=16)))
    
    legend <- g_legend(legendplot)
  }
}
margin = theme(plot.margin = unit(c(0,0,0,0), "lines"))

title1=textGrob("Plate 1 median methylated intensity", gp=gpar(fontsize = 25))
grid.arrange(nrow = 1, 
             ncol = length(plate2Meth),
             grobs = lapply(plate2Meth, "+",margin),
             top=title1)

title2=textGrob("Plate 1 median unmethylated intensity", gp=gpar(fontsize = 25))
grid.arrange(nrow = 1, 
             ncol = length(plate2UnMeth),
             grobs = lapply(plate2UnMeth, "+",margin), 
             top = title2)
```

```{r plate1 legend, echo = FALSE, warning= FALSE, message=FALSE}
plot(legend)
```

### Plate 2

```{r plate2, fig.height= 8, fig.width= 18,echo = FALSE,fig.show='hold',out.height="50%"}

samples<-QCmetrics[which(QCmetrics$PlateNumber == plates[2]),]
Sentrix <- as.character(unique(samples$Chip_ID))

plate2Meth <- list()
plate2UnMeth <- list()
for (i in 1:length(as.character(unique(samples$Chip_ID)))) {
  Sentrix_i <- Sentrix[i]
  Sample_Sentrix <- samples[which(samples$Chip_ID == Sentrix_i),]
  
  if (Sentrix_i == unique(control_sentrix)){
    control <- Sample_Sentrix[which(Sample_Sentrix$ExternalSampleID == "methylated_control"),]
    control$Control <- control$ExternalSampleID == "methylated_control"
    
    plateHeatmap <- ggplot(data=Sample_Sentrix, aes(y=Chip_Row, x=Chip_Column)) +
    scale_fill_gradientn(colours=colorRamps::matlab.like(100), limits=c(min(QCmetrics$U.median),max(QCmetrics$M.median))) +
    labs(x="", y="") +
    theme_minimal() + 
    coord_equal(clip = 'off') +
    theme(plot.title = element_text(hjust = 0.5), axis.text.x = element_text(angle = 0, hjust=1),
          axis.text.y = element_text(angle = 1, hjust=1),
          plot.margin = unit(c(0,15,0,0), "lines"))
     
  plate2Meth[[i]] <- plateHeatmap +
    geom_tile(aes(fill=M.median), colour = "white") +
    theme(legend.position = "none")  +
    geom_point(data=control, aes(y=Chip_Row, x=Chip_Column)) +
    annotate(geom = "text", x = 0.8, y = 7,
             label = Sentrix[i], hjust = 0, angle = 0, size = 5) 
  
  plate2UnMeth[[i]] <- plateHeatmap +
    geom_tile(aes(fill=U.median), colour = "white") +
    geom_point(data=control, aes(y=Chip_Row, x=Chip_Column)) +
    theme(legend.position = "none")  +
    annotate(geom = "text", x = 0.8, y = 7,
             label = Sentrix[i], hjust = 0, angle = 0, size = 5) 
  
  legendplot<-plateHeatmap + 
    geom_tile(aes(fill=U.median), colour = "white") +
    labs(fill="Intensity", alpha="Control") +
    theme_minimal() + 
    geom_point(data=control, aes(y=Chip_Row, x=Chip_Column, alpha=control$Control)) +
    scale_alpha_manual(values=c(1,1,1)) + 
    guides(alpha = guide_legend(override.aes = list(colour="black", pch=16)))
  
      legend <- g_legend(legendplot)
} 
  else
  
  {plateHeatmap <- ggplot(data=Sample_Sentrix, aes(y=Chip_Row, x=Chip_Column)) +
    scale_fill_gradientn(colours=colorRamps::matlab.like(100), limits=c(min(QCmetrics$U.median),max(QCmetrics$M.median))) +
    labs(x="", y="") +
    theme_minimal() + 
    coord_equal(clip = 'off') +
    theme(plot.title = element_text(hjust = 0.5), axis.text.x = element_text(angle = 0, hjust=1),
          axis.text.y = element_text(angle = 1, hjust=1),
          plot.margin = unit(c(0,15,0,0), "lines"))
     
  plate2Meth[[i]] <- plateHeatmap +
    geom_tile(aes(fill=M.median), colour = "white") +
    theme(legend.position = "none")  +
    annotate(geom = "text", x = 0.8, y = 7,
             label = Sentrix[i], hjust = 0, angle = 0, size = 5) 
  
  plate2UnMeth[[i]] <- plateHeatmap +
    geom_tile(aes(fill=U.median), colour = "white") +
    theme(legend.position = "none")  +
    annotate(geom = "text", x = 0.8, y = 7,
             label = Sentrix[i], hjust = 0, angle = 0, size = 5) 
    
    legendplot<-plateHeatmap + 
    geom_tile(aes(fill=U.median), colour = "white") +
    labs(fill="Intensity", alpha="Control") +
    scale_alpha_manual(values=c(1,1,1)) + 
      theme_minimal() + 
    guides(alpha = guide_legend(override.aes = list(colour="black", pch=16)))
    
    legend <- g_legend(legendplot)
  }
}
margin = theme(plot.margin = unit(c(0,0,0,0), "lines"))

title1=textGrob("Plate 2 median methylated intensity", gp=gpar(fontsize = 25))
grid.arrange(nrow = 1, 
             ncol = length(plate2Meth),
             grobs = lapply(plate2Meth, "+",margin),
             top=title1 , legend)

title2=textGrob("Plate 2 median unmethylated intensity", gp=gpar(fontsize = 25))
grid.arrange(nrow = 1, 
             ncol = length(plate2UnMeth),
             grobs = lapply(plate2UnMeth, "+",margin), 
             top = title2, legend)

```



```{r plate2 legend, echo = FALSE, warning= FALSE, message=FALSE}
plot(legend)
```





# 3  Bisulphite Conversion
A bisulphite conversion statistic for each sample was calculated, and a histogram of the results plotted. Samples with a conversion < 80% fail the QC.

```{r bisulphiteconversion, echo=FALSE}
#Make new rgset without methylated controls
RGset2<-RGset[,!sample_sheet$Control]
##New method
#green channel
csp.green <- function (RGsetEx, controls = c("BISULFITE CONVERSION I", "BISULFITE CONVERSION II"))
{
  minfi:::.isRGOrStop(RGsetEx)
  r <- getRed(RGsetEx)
  g <- getGreen(RGsetEx)
  sapply (controls, function( controlType ) {
    
    ctrlAddress <- try (getControlAddress(RGsetEx, controlType = controlType), silent = T)
    if (!inherits (ctrlAddress, 'try-error')){ctrlAddress <- getControlAddress(RGsetEx, controlType = controlType)}
    else
      stop ("450k QC data could not be found")
    
    
    g[ctrlAddress, ]
  })}
#red channel
csp.red <- function (RGsetEx, controls = c("BISULFITE CONVERSION I", "BISULFITE CONVERSION II"))
{
  minfi:::.isRGOrStop(RGsetEx)
  r <- getRed(RGsetEx)
  g <- getGreen(RGsetEx)
  sapply (controls, function( controlType ) {
    
    ctrlAddress <- getControlAddress(RGsetEx, controlType = controlType)
    
    r[ctrlAddress, ]
  })}

red <- csp.red(RGset2)
green <- csp.green(RGset2)
#selecting only the Bisulfite conversion I values from both green and red
bsI.green <- green$`BISULFITE CONVERSION I`
bsI.red <- red$`BISULFITE CONVERSION I`
#selecting only the Bisulfite conversion II values from both green and red
bsII.green <- green$`BISULFITE CONVERSION II`
bsII.red <- red$`BISULFITE CONVERSION II`

BSI.betas <- bsI.red/(bsI.red + bsI.green)
BSII.betas <- bsII.red/(bsII.red + bsII.green)
BSI.betas2 <- apply(BSI.betas,2,mean) #try to find the mean of all the probes as there is a range and this range is different to the type2 probes methylation status
Bisulphite <-apply(rbind(BSI.betas2, BSII.betas), 2, median)*100 ## this is the value you are interested in

hist(Bisulphite, xlab = "Median % BS conversion", main = "Histogram of Bisulphite Converstion Statistics")

QCmetrics <- QCmetrics[which(QCmetrics$ExternalSampleID != "methylated_control"),] 
QCmetrics$Bisulphite <- cbind(QCmetrics, Bisulphite)

```



# 4 Sex Check

We can check the sex of the samples by comparing the sex probes intensities.

``` {r, sex probes, echo = FALSE}
# Set this variable to the column name of your species of interest.
species_mappability_name = "MusMusculus"

species_probes <- probe_mapping_file[, c('probeID','MusMusculus')]
species_probes <- species_probes[!is.na(species_probes$MusMusculus),]
# Set this variable to the column name of your species of interest.
species_mappability_name = "MusMusculus"

species_probes <- probe_mapping_file[, c('probeID','MusMusculus')]
species_probes <- species_probes[!is.na(species_probes$MusMusculus),]

# The following minfi function subsets the RGChannelSet to only the probeIDs provided in a list
RGset_species <- subsetByLoci(RGset2, include=species_probes$probeID)

#filter for sex chromosomes in annotation file
sex_cpgpositions <- matrix(NA, nrow = 1, ncol = 2)
colnames(sex_cpgpositions) <- colnames(species_probes)
species_probes <- as.data.frame(species_probes)
for( i in 1:nrow(species_probes)){
  if (substring(species_probes[i,"MusMusculus"],1,1) %in% seq(1,22,1) ){
    next
  }  else{
    sex_cpgpositions <- rbind(species_probes[i,], sex_cpgpositions)
  } 
}

#susbet these X chromosomes from the RGset and plot density plot
RGset_speciessexprobes <- subsetByLoci(RGset_species, include=sex_cpgpositions$probeID)
```


```{r, SexCheck, echo=FALSE}
#Make a beta for just sex probes so association with sex is stronger when making the pca plot

####
findGenderPC<-function(betas, sex, npcs = 20){

	betas.com<-betas[complete.cases(betas),]
	pca<-prcomp(betas.com)

	pca.cor<-rep(NA, npcs)
	for(i in 1:npcs){
		pca.cor[i]<-cor(pca$rotation[,i], as.numeric(as.factor(sex)), use = "complete")
	}
	top<-order(abs(pca.cor), decreasing = TRUE)[1]
	second<-order(abs(pca.cor), decreasing = TRUE)[2]
	print(paste("Top correlated principal components with sex:", top, ",", second))

	predSex<-rep(NA, length(sex))
	options.sex<-levels(as.factor(sex))

	if(abs(pca.cor[top]) > 0.9){
		print("Top PC has r > 0.9 with sex so good enough to confirm reported sexes")
	} else {
	  print(paste("Top PC has r =", round(abs(pca.cor[top]),2), "with sex so may not be good enough to confirm reported sexes"))
	}

	if(sign(pca.cor[top]) == 1){
		predSex[which(pca$rotation[,top] < 0)]<-options.sex[1]
		predSex[which(pca$rotation[,top] > 0)]<-options.sex[2]
	} else {
		predSex[which(pca$rotation[,top] < 0)]<-options.sex[2]
		predSex[which(pca$rotation[,top] > 0)]<-options.sex[1]
	}

	plot(pca$rotation[,top], pca$rotation[,second], pch = 16, col = rainbow(3)[as.factor(sex)], xlab = paste("PC", top), ylab = paste("PC", second), main="PC Plot Coloured by Reported Sex")
	legend("bottomright", levels(as.factor(sex)), pch = 16, col = rainbow(3))

	#using mismatch to add labels to failed samples
  #mismatch<-which(predSex!=sex)
  #text(pca$rotation[mismatch,top], pca$rotation[mismatch, second], #labels=colnames(betas)[mismatch], cex=0.7, pos=4)

	return(predSex)
}


betasSexProbes <- as.matrix(getBeta(RGset_speciessexprobes), rownames="CGid")
ReportedSex<-QCmetrics$Sex #Do not pull put the methylated sexes or else matrix colum length will not match with predsex
PredictedSex<-findGenderPC(betasSexProbes,ReportedSex)

QCmetrics<-cbind(QCmetrics,PredictedSex)
QCmetrics$MismatchSex<-PredictedSex!=ReportedSex 


```



# 5 Methylation Distribution

At the moment we have probes for all 192 possible vertebrate species. Density plots of the beta values are plotted for samples with all probes and another plot of samples with specific mouse probes (separated by type I and type II probes). There are `r nrow(as.data.frame(getProbeInfo(RGset2, type="I")))` typeI probes and `r nrow(as.data.frame(getProbeInfo(RGset2, type="II")))` typeII probes.


```{r plot betas function, echo=FALSE}

densityPlot(RGset2, main= "All probes")

densityPlot(RGset_species, main= "Probes that map to mouse genome")

```

# 6 Normalisation

For normalisation, sorted samples will be normalised seperately to the the bulk samples. The normalisation will be only for the probes that map to the mouse genome. 

```{r filter, echo = FALSE, warning = FALSE, message = FALSE}
#Female - bulk, sorted - male
#Seperate RGset from the species rgset with the mouse probes

QCmetrics_bulk <- QCmetrics[which(QCmetrics$Sex == "F"),] 
RGset_bulk <- subsetByLoci(RGset_species, include=QCmetrics_bulk$Basename)
QCmetrics_sorted <-QCmetrics[which(QCmetrics$Sex == "M"),] 
RGset_sorted <-  subsetByLoci(RGset_species, include=QCmetrics_sorted$Basename)

```

```{r Normalisation, echo=FALSE}

plotRGset_density<-function(RGset, study=""){
	typeII <- as.data.frame(getProbeInfo(RGset, type="II"))
  typeI <- as.data.frame(getProbeInfo(RGset, type="I"))
  betas <- as.data.frame(getBeta(RGset), rownames="CGid")
	mat <- betas[,colnames(RGset)] 
	
	  plot(density(mat[typeI[,1],1], na.rm=T, bw=0.03), cex.main=0.8, main=paste(study, "Betas"), ylim=c(0, 5.2), xlab="")
    lines(density(mat[typeII[,1],1], na.rm=T, bw=0.03), col="red")

    for(j in 2:ncol(mat)){
		lines(density(mat[typeI[,1],j], na.rm=T, bw=0.03))
		lines(density(mat[typeII[,1],j], na.rm=T, bw=0.03), col="red")
    }
      
    legend("topright", legend=c("Type I", "Type II"), lty=1, col=c("black", "red")) 
}

#The more samples in your mset the more messy these plots will look

plotRGset_density(RGset_sorted, study="Sorted Samples (pre-normalisation)")
plotRGset_density(RGset_bulk, study="Bulk Samples (pre-normalisation)")

##Normalise samples using
RGset_sorted.dasen <- dasen(RGset_sorted) 
RGset_bulk.dasen <- dasen(RGset_bulk)

plotRGset_density(RGset_sorted.dasen, study="Sorted Samples (post-normalisation)")
plotRGset_density(RGset_bulk.dasen, study="Bulk Samples (post-normalisation)")


```



```{r summary, echo = FALSE, warning = FALSE, message = FALSE}

save(QCmetrics_bulk, RGset_bulk.dasen, file = "/gpfs/ts0/scratch/and202/Mouse/Mouse_Bulk.rdat")
save(QCmetrics_sorted, RGset_sorted.dasen, file = "/gpfs/ts0/scratch/and202/Mouse/Mouse_sorted.rdat")
```


